### ISSUE TITLE
Request with query param style `deepObject` failing validation

### ISSUE THREAD

**Issue by TristanSpeakEasy** [pb33f/wiretap#82]
This is a bit of a tough one as `deepObject` serialization is not very well defined but we support deepObject serialization that is compatible with a number of different server frameworks, but wiretap is failing validation on a request that does work with other services.

Here is the spec (btw this is a distilled down version of the real spec I am using and this distilled version causes this issue https://github.com/pb33f/wiretap/issues/81 so hard to test in isolation and I can't provide the full spec I am using atm):

```yaml
openapi: 3.1.0
info:
  title: Test
  version: 0.1.0
paths:
  /anything/queryParams/deepObject/map:
    get:
      operationId: deepObjectQueryParamsMap
      parameters:
        - name: mapParam
          in: query
          style: deepObject
          schema:
            type: object
            additionalProperties:
              type: string
            example: { "test": "value", "test2": "value2" }
          required: true
        - name: mapArrParam
          in: query
          style: deepObject
          schema:
            type: object
            additionalProperties:
              type: array
              items:
                type: string
            example: { "test": ["test", "test2"], "test2": ["test3", "test4"] }
      responses:
        "200":
          description: OK
```

The error returned is:

```json
{"type":"https://pb33f.io/wiretap/error","title":"unable to serve mocked response","status":500,"detail":"Error: Query parameter 'mapArrParam' is not a valid deepObject, Reason: The query parameter 'mapArrParam' has the 'deepObject' style defined, There are multiple values (2) supplied, instead of a single value, Line: 1498, Column: 18\nError: Query parameter 'mapArrParam' failed to validate, Reason: The query parameter 'mapArrParam' is defined as an object, however it failed to pass a schema validation, Validation Errors: [Reason: expected array, but got string, Location: /additionalProperties/type Reason: expected array, but got string, Location: /additionalProperties/type], Line: 1500, Column: 13"}
```

The query string being sent is:

```
"mapArrParam%5Btest2%5D=test3&mapArrParam%5Btest2%5D=test4&mapArrParam%5Btest%5D=test&mapArrParam%5Btest%5D=test2&mapParam%5Btest2%5D=value2&mapParam%5Btest%5D=value"
```

basically the array is represented by listing the same key multiple times

**Comment by daveshanley** [comment]
This should be resolved in `v0.1.1`
